tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -416,9 +416,9 @@ SpanBuffer SwapBuffers()
                 for (var i = 0; i < spans.Count; i++)
                 {
                     var index = i + spans.Offset;
-                    if (spans.Array![index].GetMetric(Metrics.SingleSpanSampling.SamplingMechanism) is not null)
+                    if (spans.Array?[index].GetMetric(Metrics.SingleSpanSampling.SamplingMechanism) is not null)
```

Comment:
In practice (in our codebase) this should never be able to be null if you track all the places where we create this `ArraySegment<Span>`. But who knows... 😅 🤷🏽‍♂️